src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -1,57 +1,44 @@
-import React from 'react';
-import { string, bool, number } from 'prop-types';
+import React, { useEffect } from 'react';
+import { string, shape } from 'prop-types';
 import Helmet from 'react-helmet';
 
-const CanonicalChartbeatBeacon = ({
-  domain,
-  type,
-  sections,
-  cookie,
-  chartbeatUID,
-  useCanonical,
-  chartbeatSource,
-}) => (
-  <Helmet>
-    <script async type="text/javascript">
-      {`
+const CanonicalChartbeatBeacon = ({ chartbeatConfig, chartbeatSource }) => {
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.pSUPERFLY) {
+      /*
+        This function is always called to update config values on page changes 
+        https://chartbeat.zendesk.com/hc/en-us/articles/210271287-Handling-virtual-page-changes
+      */
+      window.pSUPERFLY.virtualPage(chartbeatConfig);
+    }
+  }, [chartbeatConfig]);
+
+  return (
+    <Helmet>
+      <script async type="text/javascript">
+        {`
         (function(){
           var _sf_async_config = window._sf_async_config = (window._sf_async_config || {});
-          _sf_async_config.uid = ${chartbeatUID};
-          _sf_async_config.domain = "${domain}";
-          _sf_async_config.useCanonical = ${useCanonical};
-          _sf_async_config.useCanonicalDomain = ${useCanonical};
-          _sf_async_config.contentType = "${type}";
-          _sf_async_config.sections = "${sections}";
-          _sf_async_config.idSync = {
-           bbc_hid: "${cookie}"
-          };
-          function loadChartbeat() {
-           var e = document.createElement('script');
-           var n = document.getElementsByTagName('script')[0];
-           e.type = 'text/javascript';
-           e.async = true;
-           e.src = "${chartbeatSource}";
-           n.parentNode.insertBefore(e, n);
+          function merge(obj, src) {
+            for (var key in src) {
+                if (src.hasOwnProperty(key)) obj[key] = src[key];
+            }
+            return obj;
           }
-          loadChartbeat();
+          _sf_async_config = merge(_sf_async_config, ${JSON.stringify(
+            chartbeatConfig,
+          )});
```

Comment:
For brevity, I would get rid of the function and inline the loop. I believe we can also omit the `hasOwnProperty` as we know the config has no inherited properties. Also means we can skip the assignment on L28